### PR TITLE
archinfo: python3 compatibility

### PR DESCRIPTION
**Pull request by Mic92**
There could be more, but then something like the `future` package is needed to keep it maintainable.

### CODE REVIEW

**Comment by rhelmot** [comment]
Thanks!

/cc @tyb0807

File: archinfo/arch.py [hunk 1]
```diff
@@ -532,7 +538,7 @@ def register_arch(regexes, bits, endness, my_arch):
             raise ValueError('Invalid Regular Expression %s' % rx)
     #if not isinstance(my_arch,Arch):
     #    raise TypeError("Arch must be a subclass of archinfo.Arch")
-    if not isinstance(bits, int):
+    if not isinstance(bits, integer_types):
```
Comment (Mic92) [inline archinfo/arch.py]: mhm. I hope I did not introduce unwanted behavior here.

Comment (rhelmot) [reply archinfo/arch.py]: Shouldn't be. I'd call it a bugfix if anything.

**Comment by tyb0807** [comment]
@rhelmot Thanks for the ping. I still think that `future` package should be the most pythonic way to write Python2/3 code, even when its internal code contains a bunch of conditions as in this PR. And sorry for the late reply.

**Comment by Mic92** [comment]
If I have bigger refactoring to do in this module, I will take this into account.

**Pull request merged**